### PR TITLE
feat(api balance endpoint): add spoke balance API endpoint

### PR DESCRIPTION
**Pull request by james-a-morris**
Dependent on @daywiss's #81 endpoint. Adds spoke balance tracking.

### CODE REVIEW

File: packages/indexer-api/src/dtos/balances.dto.ts [hunk 1]
```diff
@@ -2,15 +2,25 @@ import * as s from "superstruct";
...
-  l1Token: s.string(),
+  l1Token: s.optional(s.string()),
 });
 
 // query spokepools by chainId, must specify
 export const SpokePoolBalanceParams = s.object({
   chainId: s.number(),
+  l1Token: s.optional(s.string()),
   // unsure why we have timestamp, implies we are storign history of balances? this is in the spec.
   timestamp: s.number(),
```
Comment (melisaguevara) [inline packages/indexer-api/src/dtos/balances.dto.ts:13]: Seems like we are not using this, should we delete it?

**Review by melisaguevara — COMMENTED** [review]
LGTM! Just one question

File: packages/indexer-api/src/dtos/balances.dto.ts (hunk 1 above)
Comment (james-a-morris) [reply packages/indexer-api/src/dtos/balances.dto.ts:13]: It's in the spec so I'm inclined to keep it. cc @amateima

File: packages/indexer-api/src/dtos/balances.dto.ts [hunk 2]
```diff
@@ -8,8 +8,19 @@ export const HubPoolBalanceQueryParams = s.object({
 // query spokepools by chainId, must specify
 export const SpokePoolBalanceParams = s.object({
   chainId: s.number(),
+  l1Token: s.optional(s.string()),
   // unsure why we have timestamp, implies we are storign history of balances? this is in the spec.
-  timestamp: s.number(),
-  // unsure why specified as l2Token in spec, don't we have spoke pool on L1?
-  l2Token: s.optional(s.number()),
+  timestamp: s.optional(s.number()),
```
Comment (amateima) [inline packages/indexer-api/src/dtos/balances.dto.ts:13]: we can remove this for now

**Review by amateima — APPROVED** [review]

File: packages/indexer-api/src/dtos/balances.dto.ts [hunk 3]
```diff
@@ -8,8 +8,19 @@ export const HubPoolBalanceQueryParams = s.object({
 // query spokepools by chainId, must specify
 export const SpokePoolBalanceParams = s.object({
```
Comment (daywiss) [inline packages/indexer-api/src/dtos/balances.dto.ts:9]: one nit here was to rename this to `SpokePoolBalanceQueryParams`

**Review by daywiss — APPROVED** [review]
pretty cool

**Review by amateima — APPROVED** [review]

File: packages/indexer-api/src/dtos/balances.dto.ts (hunk 3 above)
Comment (james-a-morris) [reply packages/indexer-api/src/dtos/balances.dto.ts]: Done

**Pull request merged**